code/lib/core-common/src/utils/normalize-path.ts

```diff
@@ -0,0 +1,14 @@
+import path from 'path';
+
+/**
+ * Normalize a path to use forward slashes and remove .. and .
+ * @param p The path to normalize
+ * @returns The normalized path
+ * @example
+ * normalizePath('path/to/../file') // => 'path/file'
+ * normalizePath('path/to/./file') // => 'path/to/file'
+ * normalizePath('path\\to\\file') // => 'path/to/file'
+ */
+export function normalizePath(p: string) {
+  return path.normalize(p).replace(/\\/g, '/');
```

Comment:
I think we should replace `\\` before normalizing. Otherwise you can do `foo\\..\\bar` and it would still yield `foo/../bar`.